crates/stateless/src/chain_spec.rs

```diff
@@ -0,0 +1,122 @@
+use core::fmt::Display;
+
+use alloc::{boxed::Box, vec::Vec};
+use alloy_consensus::Header;
+use alloy_eips::BlobScheduleBlobParams;
+use alloy_genesis::Genesis;
+use alloy_primitives::{Address, B256, U256};
+use reth_chainspec::{
+    BaseFeeParams, Chain, ChainHardforks, DepositContract, EthChainSpec, EthereumHardfork,
+    EthereumHardforks, ForkCondition, ForkFilter, ForkId, Hardfork, Hardforks, Head,
+};
+use reth_evm::eth::spec::EthExecutorSpec;
+
+#[derive(Debug, Clone)]
+pub struct ChainSpec {
+    pub chain: Chain,
+    pub hardforks: ChainHardforks,
```

Comment:
Prefered using this native type, compared to [zeth variant](https://github.com/risc0/zeth/blob/main/crates/chainspec/src/lib.rs#L73).

This helps avoid complexity on what we do in `crates/stateless/src/fork_spec.rs`, since we can use the `hardforks` generated from the `ChainSpec` crate directly.